README.md

```diff
@@ -84,29 +93,58 @@ Select row subsets. Operates on both a `DataFrame` and a `GroupedDataFrame`.
 ```julia
 df = DataFrame(x = [1, 1, 2, 2], y = [1, 2, 101, 102]);
 gd = groupby(df, :x);
-x = 1;
+outside_var = 1;
 @where(df, :x .> 1)
-@where(df, :x .> x)
-@where(df, :x .> x, :y .< 102)  # the two expressions are "and-ed"
+@where(df, :x .> outside_var)
+@where(df, :x .> outside_var, :y .< 102)  # the two expressions are "and-ed"
 @where(gd, :x .> mean(:x))
 ```
 
 ## `@combine`
 
-Summarize or collapse, a grouped data frame by perforing transformations at the group level and 
+Summarize, or collapse, a grouped data frame by performing transformations at the group level and 
 collecting the result into a single data frame. Also works on a `DataFrame`, which 
 acts like a `GroupedDataFrame` with one group. 
 
-Does not allow input in first argument like `combine` from DataFrames.jl. But an argument
-returning a `Table`-like object can be input for the first argument. 
+Requires a `DataFrame` or `GroupedDataFrame` as the first argument, unlike 
+`combine` from DataFrames.jl. For example, the following code runs:
+
+```
+df = DataFrame(x = [1, 1, 2, 2], y = [1, 2, 101, 102]);
+gd = groupby(df, :x);
+combine(d -> (a = sum(d.x), b = sum(d.y)), gd)
+```
+
+However `@combine((a = sum(:x), b = sum(:y)), gd)` will fail because `@combine` requires a 
+`GroupedDataFrame` or a `DataFrame` as the first argument. The following is equivalent
+to the `combine` call above
+
+```
+df = DataFrame(x = [1, 1, 2, 2], y = [1, 2, 101, 102]);
+gd = groupby(df, :x);
+@combine(gd, (a = sum(:x), b = sum(:y)))
+```
+
+For arguments which return a table-like object, such as `(a = sum(:x), b = sum(:y))`, above
+`@combine` only allows *one* argument and it must be the *second* positional argument. 
+Consider the call 
+
+```
+@combine(gd, (a = sum(:x), b = sum(:y)), c = first(:x))
+```
+
+the above will fail because `@combine` does not accept a "keyword argument"-style column 
+creation after a "return a table"-style column creation call. 
+
+Examples:
```

Comment:
I'd show these examples before the paragraphs above that describe what doesn't work. Otherwise you need to read a lot of text about corner cases before seeing simple examples of expected usage.